plugins/available/base.plugin.bash

```diff
@@ -4,13 +4,24 @@ about-plugin 'miscellaneous tools'
 function ips ()
 {
     about 'display all ip addresses for this host'
+    param 'optional -i'
+    example '$ ips'
+    example '$ ips -i'
     group 'base'
     if command -v ifconfig &>/dev/null
     then
-        ifconfig | awk '/inet /{ gsub(/addr:/, ""); print $2 }'
+        if [ -n "$1" -a "$1" = "-i" ];then
+            ifconfig | grep -B 1 -E "inet "| sed 's/addr://g' | sed 's/: / /g' | awk '{if (NR%3==1) print $1 ":"; else if (NR%3==2) print $2}'
+        else
+            ifconfig | awk '/inet /{ gsub(/addr:/, ""); print $2 }'
+        fi
     elif command -v ip &>/dev/null
     then
-        ip addr | grep -oP 'inet \K[\d.]+'
+        if [ -n "$1" -a "$1" = "-i" ];then
+            ip addr | grep -E '^\s*[1-9]| (25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[1-9][0-9]|[1-9])\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})' | awk '{print $2}'
```

Comment:
I will added a code comment. 
ip addr --- running ip addr to display all addresses or other on the device.
the result:
.1: lo: <LOOPBACK,UP,LOWER_UP> mtu 16436 qdisc noqueue state UNKNOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
    inet 3.3.3.29/24 brd 3.3.3.255 scope global lo:1
    inet 3.3.3.30/24 brd 3.3.3.255 scope global secondary lo:2
    inet6 ::1/128 scope host 
   valid_lft forever preferred_lft forever
21: gre0: <NOARP> mtu 1476 qdisc noop state DOWN 
    link/gre 2.2.2.2 brd 2.2.2.1

grep command --- filter line 1 and 3,4,5 in lo device list. if in gre device, filter line 1 and 2.
awk command  --- print lo or ip address.

the result of the ips -i:
lo:
127.0.0.1/8
3.3.3.29/24
3.3.3.30/24
gre0:
2.2.2.2